persistence/src/main/java/com/tapptitude/persistence/di/DatabaseModule.kt

```diff
@@ -0,0 +1,31 @@
+package com.tapptitude.persistence.di
+
+import android.app.Application
+import androidx.room.Room
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import com.tapptitude.persistence.LocalDatabase
+import com.tapptitude.persistence.images.ImagesDao
+import org.koin.android.ext.koin.androidApplication
+import org.koin.dsl.module
+
+const val DATABASE_NAME = "LocalStore.db"
+
+val databaseModule = module {
+    fun provideDatabase(application: Application): LocalDatabase =
+        Room.databaseBuilder(application, LocalDatabase::class.java, DATABASE_NAME)
+            .addMigrations(object : Migration(1, 2) {
```

Comment:
Is it ok to have both a `Migration` and `fallbackTodestructiveMigration()`.
I honestly don't how it will behave. Will Room migrate what knows and resets the rest?

Reply:
AFAIK if it finds some inconsistency between versioning and no migration provided it will reset everything, so for example at some version you forget to specify Migration (3, 4) then instead of crashing your app it will just reset the database. It is just a fallback, so a fail-safe

Reply:
Nice, did not know that. Thanks for fyi